src/keypair.ts

```diff
@@ -123,4 +127,64 @@ export class Keypair {
   verify(data: Buffer, signature: Buffer) {
     return verify(data, signature, this.pubKey)
   }
+
+  /**
+   * Encrypts the given `data` using a Curve25519 'variant' of the public key.
+   *
+   * The encryption uses an ephemeral private key, which is prepended to the ciphertext,
+   * along with a nonce of random bytes.
+   *
+   * @note See https://github.com/dchest/ed2curve-js for conversion details.
+   * @param data Data to encrypt
+   */
+  encrypt(data: Buffer) {
```

Comment:
This is essentially a port of our `go-textile/crypto/asymmetric.go` `Encrypt` and `Decrypt` functions. We use `ed2curve` here, which is apparently compatible with `github.com/agl/ed25519/extra25519` (which is what we use).

Reply:
wow, nice